### PR TITLE
Deepgram transcriber

### PR DESCRIPTION
**Pull request by alexis-kochetov**

### CODE REVIEW

File: src/dotnet/Audio.Client/AudioClient.cs [hunk 1]
```diff
@@ -35,6 +35,7 @@ public AudioClient(IServiceProvider services)
             .Select((data, i) => new AudioFrame {
                 Data = data,
                 Offset = TimeSpan.FromMilliseconds(i * 20), // we support only 20-ms packets
+                Duration = TimeSpan.FromMilliseconds(20),
```
Comment (frolyo) [inline src/dotnet/Audio.Client/AudioClient.cs]: probably 20ms can be extracted to a constant and offset will be i * const

File: src/dotnet/Audio.Service/AudioHub.cs [hunk 2]
```diff
@@ -79,6 +79,7 @@ await foreach (var chunk in stream.ConfigureAwait(false))
             .Select((packet, i) => new AudioFrame {
                 Data = packet,
                 Offset = TimeSpan.FromMilliseconds(i * 20), // we support only 20-ms packets
+                Duration = TimeSpan.FromMilliseconds(20),
```
Comment (frolyo) [inline src/dotnet/Audio.Service/AudioHub.cs]: and will be used used here as well

File: src/dotnet/Audio.Service/AudioStreamer.cs [hunk 3]
```diff
@@ -33,6 +33,7 @@ public class AudioStreamer : IAudioStreamer
             .Select((data, i) => new AudioFrame {
                 Data = data,
                 Offset = TimeSpan.FromMilliseconds(i * 20), // we support only 20-ms packets
+                Duration = TimeSpan.FromMilliseconds(20),
```
Comment (frolyo) [inline src/dotnet/Audio.Service/AudioStreamer.cs]: and here

File: src/dotnet/Audio/ActualOpusStreamConverter.cs [hunk 4]
```diff
@@ -82,6 +82,7 @@ static void ReadFrames(ref ReadOnlySequence<byte> sequence, List<AudioFrame> fra
                                 frames1.Add(new AudioFrame {
                                     Data = packet,
                                     Offset = TimeSpan.FromMilliseconds(offsetMs),
+                                    Duration = TimeSpan.FromMilliseconds(20),
```
Comment (frolyo) [inline src/dotnet/Audio/ActualOpusStreamConverter.cs]: and here

File: src/dotnet/Audio/Ogg/OggCRC32.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        0xEA23F0AF, 0xEEE2ED18, 0xF0A5BD1D, 0xF464A0AA, 0xF9278673, 0xFDE69BC4, 0x89B8FD09, 0x8D79E0BE, 0x803AC667,
+        0x84FBDBD0, 0x9ABC8BD5, 0x9E7D9662, 0x933EB0BB, 0x97FFAD0C, 0xAFB010B1, 0xAB710D06, 0xA6322BDF, 0xA2F33668,
+        0xBCB4666D, 0xB8757BDA, 0xB5365D03, 0xB1F740B4,
+    };
+
+    public static uint Get(uint crc, ReadOnlySpan<byte> data)
+    {
+        ReadOnlySpan<uint> crcTable = _crcTable;
+        // // ReSharper disable once ForCanBeConvertedToForeach
+        // for (int i = 0; i < data.Length; i++)
```
Comment (frolyo) [inline src/dotnet/Audio/Ogg/OggCRC32.cs]: do you need this code?

File: src/dotnet/Audio/Ogg/OggHeader.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        sizeof(uint)
+        + sizeof(byte)
+        + sizeof(OggHeaderTypeFlag)
+        + sizeof(ulong)
+        + sizeof(uint)
+        + sizeof(int)
+        + sizeof(uint)
+        + sizeof(byte)
+        + (SegmentTable.Length * sizeof(byte));
+
```
Comment (frolyo) [inline src/dotnet/Audio/Ogg/OggHeader.cs]: redundant spaces

File: src/dotnet/Audio/Ogg/OggPage.cs
Comment (frolyo) [inline src/dotnet/Audio/Ogg/OggPage.cs:1]: Is it used at all?

File: src/dotnet/Transcription.Service/Deepgram/DeepgramTranscriber.cs [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+                "es-MX" => "es-419",
+                "es-US" => "es-419",
+                "ru-RU" => "ru",
+                "zh-CN" => "zh-CN",
+                _ => throw StandardError.NotSupported(typeof(DeepGramTranscriber), $"Language '{options1.Language.Id}' is not supported"),
+            };
+        }
+
+        [SuppressMessage("ReSharper", "AccessToDisposedClosure")]
+        void HandleConnectionOpened(object? sender, ConnectionOpenEventArgs e)
```
Comment (frolyo) [inline src/dotnet/Transcription.Service/Deepgram/DeepgramTranscriber.cs:118]: expression bodies

File: src/dotnet/Transcription.Service/Deepgram/DeepgramTranscriber.cs [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+            mapPoints.Add(veryLastPoint);
+        timeMap = new LinearMap(mapPoints.ToArray()).Simplify(Transcript.TimeMapEpsilon);
+        return true;
+    }
+
+    private static string FixSuffix(string prefix, string suffix)
+    {
+        /*
+        // Trim trailing whitespace
+        var lastLetterIndex = suffix.Length - Transcript.ContentStartRegex.Match(suffix).Length;
```
Comment (frolyo) [inline src/dotnet/Transcription.Service/Deepgram/DeepgramTranscriber.cs]: in cause it's a useful code can you leave a comment when to use it?

File: tests/Transcription.IntegrationTests/DeepgramTranscriberTest.cs [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+namespace ActualChat.Transcription.IntegrationTests;
+
+public class DeepgramTranscriberTest(ITestOutputHelper @out, ILogger<DeepGramTranscriber> log): TestBase(@out)
+{
+    private ILogger<DeepGramTranscriber> Log { get; } = log;
+
+    [Theory(Skip = "Manual")]
+    // [InlineData("file.webm", false)]
+    [InlineData("file.webm", true)]
+    // [InlineData("0002-AK.opuss", true)]
```
Comment (frolyo) [inline tests/Transcription.IntegrationTests/DeepgramTranscriberTest.cs:20]: why are these cases off?

**Review by frolyo — CHANGES_REQUESTED** [review]

File: tests/Transcription.IntegrationTests/DeepgramTranscriberTest.cs (hunk 9 above)
Comment (alexis-kochetov) [reply tests/Transcription.IntegrationTests/DeepgramTranscriberTest.cs:20]: This is a manual only test - so other options are useful for manual checks

**Pull request merged**